### PR TITLE
(PDB-2838) server side changes for remediation field

### PR DESCRIPTION
**Pull request by wkalt**
Add support for the new "remediation" field in the /reports and /nodes
responses, along with corresponding changes for the benchmark tool.

### CODE REVIEW

File: src/puppetlabs/puppetdb/scf/migrate.clj [hunk 1]
```diff
@@ -1062,6 +1062,14 @@
     "ALTER TABLE reports ADD COLUMN noop_pending boolean"
     "CREATE INDEX idx_reports_noop_pending on reports using btree (noop_pending) where (noop_pending = true)"))
 
+(defn add-remediation-columns
+  []
+  (jdbc/do-commands
+    "alter table reports add column remediation boolean"
+    "alter table resource_events add column remediation boolean"
+    "create index idx_reports_remediation on reports(remediation)"
```
Comment (senior) [inline src/puppetlabs/puppetdb/scf/migrate.clj]: Are we sure this is needed?

File: src/puppetlabs/puppetdb/scf/storage.clj [hunk 2]
```diff
@@ -1199,12 +1199,19 @@
       #(not (or (empty? %) (kitchensink/string-contains? "[" %)))
       (reverse containment-path)))))
 
-(def store-resources-column? (atom false))
-(defn maybe-resources
-  [row-map]
-  (if @store-resources-column?
+(def enterprise-data? (atom false))
```
Comment (senior) [inline src/puppetlabs/puppetdb/scf/storage.clj]: Why are we not storing this boolean?

Comment (wkalt) [inline src/puppetlabs/puppetdb/scf/storage.clj]: in the database? Not sure why we would. For what it's worth this is just a generalization of the atom that was already there; just wanted to change it not to be resource-specific.

**Comment by wkalt** [comment]
@senior I removed the index you pointed to because I couldn't get a query to use it -- it might be dependent on the proportion of true/false values though, so we might want to keep any eye on it.

File: src/puppetlabs/puppetdb/reports.clj [hunk 3]
```diff
@@ -66,17 +68,21 @@
...
  (-> report-wireformat-schema
-     (dissoc :producer :noop_pending)))
+     (dissoc :producer :noop_pending :corrective_change)))
 
 (def report-v6-wireformat-schema
   (-> report-v7-wireformat-schema
       (dissoc :catalog_uuid :cached_catalog_status :code_id)))
 
-(def resource-event-v5-wireformat-schema
+(def resource-event-v8-wireformat-schema
```
Comment (ajroetker) [inline src/puppetlabs/puppetdb/reports.clj]: There are no resource-events in the v8 wireformat, only resources

File: src/puppetlabs/puppetdb/scf/storage.clj (hunk 2 above)
Comment (ajroetker) [inline src/puppetlabs/puppetdb/scf/storage.clj]: Fwiw @mullr and I talked about this quite a bit and settled on *not* grouping all these things together.

File: src/puppetlabs/puppetdb/reports.clj (hunk 3 above)
Comment (wkalt) [inline src/puppetlabs/puppetdb/reports.clj]: those contain resource events though right?

**Pull request merged**